src/upnp.cpp

```diff
@@ -965,8 +965,31 @@ void upnp::on_upnp_xml(error_code const& e
 		return;
 	}
 
+	// determine IGD version
+	std::string igd_str(p.get_body().begin, p.get_body().end);
```

Comment:
> looking at this code, doesn't it look a bit backwards? There is an xml parser below, so why just do a string-match search first?

I do see there is a xml parser down there, yet I am reluctant to use it, because either I need to modify `find_control_url` (which is already doing a lot of work and hard to hack in there), or I need to create another callback for this simple job.

I am not sure what you suggest, but it seems writing another cb is inevitable !?